### PR TITLE
Patch for compatibility with openalpr-ios 2_0

### PR DESCRIPTION
**Pull request by twelve17**
Hi there,

I am not sure if this is the best approach for what I am trying to do, so feedback is welcome.  

For working with the 2.2.x version of alpr and later (which uses opencv 3), I have refactored the `openalpr-ios` project to use a script that will create a universal static iOS openalpr.framework bundle, and will include tesseract, leptonica, and opencv (3) frameworks (as opposed to the current process, which is partially manual).  The script will build openalpr with the aid of a [iOS toolchain file](https://github.com/twelve17/openalpr-ios/blob/2_0/etc/cmake/Toolchains/iOS.cmake), which, among other things, will set the `IOS` CMake variable to `TRUE`.

Along with the above, I am proposing these changes to a couple of the openalpr CMakeLists.txt files to help the two parts work together.

Basically, the changes are:
- Allow passing in `Tesseract_IOS_FRAMEWORK_PATH`, `Leptonica_IOS_FRAMEWORK_PATH`, and `OpenCV_IOS_FRAMEWORK_PATH` CMake cache values that alter how openalpr finds the headers and libraries for these dependencies.
- Omit build of the `alpr` executable and installation man files when `IOS` is true
- Build `statedetection` as a static library when `IOS` is true

You can see the updated `openalpr-ios` changes [on the `2_0` branch](https://github.com/twelve17/openalpr-ios/tree/2_0) if you are interested.

### CODE REVIEW

**Comment by matthill** [comment]
Alex,

Thanks for submitting this pull request.  Sorry for the delay.  This looks like a great solution.  

I'm wondering if it would make sense to get rid of the IOS from "Tesseract_IOS_FRAMEWORK_PATH"  "Leptonica_IOS_FRAMEWORK_PATH" and "OpenCV_IOS_FRAMEWORK_PATH" (e.g., Tesseract_FRAMEWORK_PATH).  I imagine this could be useful for any platform (not just iOS) if you want to hard-code a location for these libraries and not mess with the CMake find scripts.

**Comment by twelve17** [comment]
Hi Matt,

I think you're right.  I was originally afraid that perhaps there were non iOS/OSX frameworks I was unaware of, and didn't want to be too generic, but from what I'm seeing in the cmake documentation, the framework concept seems to be macos/ios specific.  I've updated this PR with this change.

Cheers,

Alex
